apisix/utils/batch-processor-manager.lua

```diff
@@ -94,11 +108,17 @@ function _M:add_entry(conf, entry)
     end
 
     log_buffer:push(entry)
+    self.total_pushed_entries = self.total_pushed_entries + 1
     return true
 end
 
 
-function _M:add_entry_to_new_processor(conf, entry, ctx, func)
+function _M:add_entry_to_new_processor(conf, entry, ctx, func, max_pending_entries)
+    if max_pending_entries and
+       self.total_pushed_entries - total_processed_entries(self) > max_pending_entries then
+        core.log.error("max pending entries limit exceeded. discarding entry")
```

Comment:
need to print more useful information, eg: max_pending_entries, `self.total_pushed_entries - total_processed_entries(self)`